--- origsrc/clisp-2.49/modules/fastcgi/fastcgi_wrappers.c	2007-06-05 11:05:46.000000000 -0400
+++ src/clisp-2.49/modules/fastcgi/fastcgi_wrappers.c	2015-02-22 10:08:04.853165900 -0500
@@ -38,6 +38,8 @@ extern char **environ;
 
 #include "fcgi_stdio.h"
 
+#include <stdlib.h>		/* For malloc */
+
 /* Crank this up as needed */
 #define TEMPBUFSIZE 65536
 
--- origsrc/clisp-2.49/modules/rawsock/rawsock.c	2010-02-17 18:39:33.000000000 -0500
+++ src/clisp-2.49/modules/rawsock/rawsock.c	2015-02-22 10:02:05.434608400 -0500
@@ -10,6 +10,8 @@
 #include "clisp.h"
 
 #include "config.h"
+/* configure glitch */
+#undef HAVE_SOCKATMARK
 
 #if defined(HAVE_SYS_TIME_H)
 # include <sys/time.h>
--- origsrc/clisp-2.49/modules/syscalls/calls.c	2015-02-21 22:39:19.335481500 -0500
+++ src/clisp-2.49/modules/syscalls/calls.c	2015-02-22 09:38:32.041766900 -0500
@@ -615,6 +615,9 @@ DEFUN(POSIX:STRING-TIME, format &optiona
   } else error_string_integer(STACK_1);
 }
 
+/* Gnulib glitch; prototype of strverscmp is missing. */
+int strverscmp (const char *s1, const char *s2);
+
 /* ========================== string comparison ========================== */
 /* call strverscmp() on STACK_0 & STACK_1 and remove them from STACK */
 static /*maygc*/ int string_version_compare (void) {
