--- origsrc/clisp-2.49/tests/socket.tst	2008-11-25 12:00:59.000000000 -0500
+++ src/clisp-2.49/tests/socket.tst	2015-02-20 11:54:55.913178400 -0500
@@ -517,7 +517,8 @@ T
 (socket:socket-connect 12345 "localhost" :timeout 30) ERROR ; ECONNREFUSED
 (open-stream-p (setq *socket-1* (socket:socket-connect
                                  12345 "localhost" :timeout 0))) T
-(read-line *socket-1*) ERROR ; ECONNREFUSED
+;; cygwin waits here
+#-cygwin(read-line *socket-1*) #-cygwin ERROR ; ECONNREFUSED
 (close *socket-1*) T
 
 (let ((interfaces '(nil "localhost" "0.0.0.0" "127.0.0.1")))
