--- origsrc/clisp/src/unix.d	2015-02-23 15:10:57.383267400 -0500
+++ src/clisp/src/unix.d	2015-02-23 16:37:05.242852000 -0500
@@ -433,6 +433,25 @@ extern_C int isatty (int fd); /* TTYNAME
 #elif defined(NEED_SYS_IOCTL_H)
   #include <sys/ioctl.h>
 #endif
+
+/* The following #ifdef block was removed as part of rev 15470; but
+then ioctl isn't declared on Cygwin, leading to an
+implicit-function-declaration error in unixaux.d */
+
+#ifdef HAVE_IOCTL
+  #ifdef IOCTL_DOTS
+    extern_C int ioctl (int fd, IOCTL_REQUEST_T request, ...); /* IOCTL(2) */
+    #define IOCTL_ARGUMENT_T  CADDR_T
+  #else
+    extern_C int ioctl (int fd, IOCTL_REQUEST_T request, IOCTL_ARGUMENT_T arg); /* IOCTL(2) */
+    /* 3rd argument is always cast to type IOCTL_ARGUMENT_T (usually CADDR_T): */
+    #define ioctl(fd,request,arg)  (ioctl)(fd,request,(IOCTL_ARGUMENT_T)(arg))
+  #endif
+#endif
+#ifndef HAVE_SELECT
+/* fcntl() will be used, declared in <fcntl.h> */
+#endif
+
 /* START_NO_BLOCK() & END_NO_BLOCK() should appear in pairs
    inside { NO_BLOCK_DECL(); ... };
  NO_BLOCK_DECL() should be before the first statement,
