--- origsrc/clisp-2.49/modules/gtk2/gtk-server.cfg	2008-05-30 17:30:44.000000000 -0400
+++ src/clisp-2.49/modules/gtk2/gtk-server.cfg	2015-02-20 11:54:55.883176700 -0500
@@ -30,14 +30,14 @@
 # GDK_LIB_NAME = libgdk.so
 # GLIB_LIB_NAME = libglib.so
 #
-# GTK2 backend:
-GTK_LIB_NAME = libgtk-x11-2.0.so
-GDK_LIB_NAME = libgdk-x11-2.0.so
-GLIB_LIB_NAME = libglib-2.0.so
-GOBJECT_LIB_NAME = libgobject-2.0.so
-ATK_LIB_NAME = libatk-1.0.so
-PANGO_LIB_NAME = libpango-1.0.so
-PIXBUF_LIB_NAME = libgdk_pixbuf_xlib-2.0.so
+# GTK2 backend cygwin:
+GTK_LIB_NAME = cyggtk-x11-2.0-0.dll
+GDK_LIB_NAME = cyggdk-x11-2.0-0.dll
+GLIB_LIB_NAME = cygglib-2.0-0.dll
+GOBJECT_LIB_NAME = cyggobject-2.0-0.dll
+ATK_LIB_NAME = cygatk-1.0-0.dll
+PANGO_LIB_NAME = cygpango-1.0-0.dll
+PIXBUF_LIB_NAME = cyggdk_pixbuf_xlib-2.0-0.dll
 #
 # XForms backend:
 # FORMS_LIB_NAME = libforms.so
