--- origsrc/clisp/modules/syscalls/configure	2015-02-23 14:46:53.000000000 -0500
+++ src/clisp/modules/syscalls/configure	2015-02-23 15:06:56.149469600 -0500
@@ -4798,7 +4798,7 @@ fi
 done
 
 if test "$ac_cv_header_shlobj_h" = yes ; then
-   LIBS=${LIBS}' -luser32 -lole32 -loleaut32 -luuid -lversion';
+   LIBS=${LIBS}' -luser32 -lole32 -loleaut32 -L/usr/lib/w32api -luuid -lversion';
 fi
 # Check whether --enable-largefile was given.
 if test "${enable_largefile+set}" = set; then :
