--- origsrc/clisp/modules/postgresql/test.tst	2015-03-10 21:32:49.000000000 -0400
+++ src/clisp/modules/postgresql/test.tst	2015-03-18 20:40:04.564711400 -0400
@@ -22,6 +22,9 @@
 
 (defparameter *trace* (os:fopen "postgres.log" "w")) *trace*
 
+;; Get "could not connect to server" on Cygwin.  Maybe need to have
+;; postgresql service running?
+#-cygwin
 (sql:with-sql-connection (conn :name "template1" :log *standard-output*)
   (sql:PQtrace conn *trace*)
   (sql:sql-transaction conn "BEGIN" sql:PGRES_COMMAND_OK)
@@ -51,7 +54,7 @@
   (sql:sql-transaction conn "COMMIT" sql:PGRES_COMMAND_OK)
   (sql:PQuntrace conn)
   NIL)
-NIL
+#-cygwin NIL
 
 (os:fclose *trace*) NIL
 (integerp (show (finish-file "postgres.log"))) T
