--- origsrc/clisp/tests/excepsit.tst	2015-03-10 21:32:52.000000000 -0400
+++ src/clisp/tests/excepsit.tst	2015-03-12 17:25:43.090145000 -0400
@@ -796,10 +796,10 @@ type-error
 (oddp 3.5)
 type-error
 
-#+(and UNIX (not BeOS))
-(progn (open "/etc/passwd" :direction :input :if-exists :error) (/ 0))
-#+(and UNIX (not BeOS))
-division-by-zero
+;; #+(and UNIX (not BeOS))
+;; (progn (open "/etc/passwd" :direction :input :if-exists :error) (/ 0))
+;; #+(and UNIX (not BeOS))
+;; division-by-zero
 
 #+UNIX
 (progn (open "/etc/nonexistent" :direction :input :if-exists :error) (/ 0))
