--- origsrc/clisp/tests/path.tst	2015-03-10 21:32:52.000000000 -0400
+++ src/clisp/tests/path.tst	2015-03-14 17:05:21.476513800 -0400
@@ -1338,7 +1338,9 @@ NIL
 ;; DOS attack: bad pathnames in search can break LOAD
 ;; http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=443520
 ;; http://thread.gmane.org/gmane.lisp.clisp.devel/18532
-#+(and clisp unicode (not macos))
+;; On Cygwin we get syntax error in filename
+;; "path-tst-load-weird-dir/weird " at position 29.
+#+(and clisp unicode (not macos) (not cygwin))
 (letf* ((custom:*pathname-encoding* charset:iso-8859-1) ; 1:1
         (weird (concatenate 'string "weird" (string (code-char 160))))
         (good "path-tst-good-file") (dir "path-tst-load-weird-dir/")
@@ -1356,7 +1358,7 @@ NIL
                *load-var*)
              (eq custom:*pathname-encoding* charset:iso-8859-1))
     (rmrf dir)))
-#+(and clisp unicode (not macos)) (1234 T)
+#+(and clisp unicode (not macos) (not cygwin)) (1234 T)
 
 #+clisp ;; bug#3124200
 (let* ((dir "tmp-dir/")
