--- origsrc/clisp/src/unix.d	2017-03-03 11:12:07.000000000 -0500
+++ src/clisp/src/unix.d	2017-03-03 12:32:35.876086900 -0500
@@ -424,7 +424,7 @@ extern_C int isatty (int fd); /* TTYNAME
 #endif
 #if defined(NEED_SYS_FILIO_H)
   #include <sys/filio.h>
-#elif defined(NEED_SYS_IOCTL_H)
+#elif defined(HAVE_IOCTL)
   #include <sys/ioctl.h>
 #endif
 /* START_NO_BLOCK() & END_NO_BLOCK() should appear in pairs
