--- origsrc/clisp/tests/path.tst	2017-03-24 13:11:59.000000000 -0400
+++ src/clisp/tests/path.tst	2017-03-24 13:19:42.785828700 -0400
@@ -1349,7 +1349,9 @@ NIL
 ;; http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=443520
 ;; http://thread.gmane.org/gmane.lisp.clisp.devel/18532
 ;; https://sourceforge.net/p/clisp/mailman/message/19813221/
-#+(and clisp unicode (not macos))
+;; On Cygwin we get syntax error in filename
+;; "path-tst-load-weird-dir/weird " at position 29.
+#+(and clisp unicode (not macos) (not cygwin))
 (letf* ((custom:*pathname-encoding* charset:iso-8859-1) ; 1:1
         (weird (concatenate 'string "weird" (string (code-char 160))))
         (good "path-tst-good-file") (dir "path-tst-load-weird-dir/")
@@ -1367,7 +1369,7 @@ NIL
                *load-var*)
              (eq custom:*pathname-encoding* charset:iso-8859-1))
     (rmrf dir)))
-#+(and clisp unicode (not macos)) (1234 T)
+#+(and clisp unicode (not macos) (not cygwin)) (1234 T)
 
 #+clisp ;; https://sourceforge.net/p/clisp/bugs/579/
 (let* ((dir "tmp-dir/")
