--- origsrc/clisp/modules/berkeley-db/test.tst	2017-03-03 11:11:57.000000000 -0500
+++ src/clisp/modules/berkeley-db/test.tst	2017-03-07 14:27:30.600872300 -0500
@@ -78,7 +78,8 @@ NIL
 (bdb:db-put *db* "foo" "bar")   NIL
 (bdb:db-put *db* "fep" "blicket")  NIL
 
-(car (show (multiple-value-list (bdb:db-compact *db*)))) NIL
+(car (show (multiple-value-list (bdb:db-compact *db*))))
+#-cygwin NIL #+cygwin #(101 112)
 
 (bdb:db-sync *db*) NIL
 (show-db *db*) NIL
