diff -r 4c9d0558e61e src/makemake.in
--- a/src/makemake.in	Sun Feb 19 16:45:35 2017 +0100
+++ b/src/makemake.in	Tue Mar 21 12:37:38 2017 -0400
@@ -3026,7 +3026,7 @@
 # clx depends on the symbol "hostent_to_lisp", which is defined in the
 # syscalls module and therefore doesn't get put into lisp.def on
 # Cygwin.  So we just force it.
-if [ "${HSYSOS}" != cygwin ]; then
+if [ "${HSYSOS}" != cygwin -o "${with_dynamic_modules}" = no ]; then
   echotab "(echo; echo '/* genclisph */' ; ${HERE}genclisph clisp-test.c ${CLISP_DEF}; echo ; echo '#endif /* _CLISP_H */') >> clisp.h"
 else
     echotab "(echo; echo '/* genclisph */' ; ${HERE}genclisph clisp-test.c ${CLISP_DEF}; echo -e \"\tlisp.exe.hostent_to_lisp\" >> ${CLISP_DEF} ; echo ; echo '#endif /* _CLISP_H */') >> clisp.h"
