--- origsrc/clisp-2.48/src/makemake.in	2015-02-18 19:50:02.109044900 -0500
+++ src/clisp-2.48/src/makemake.in	2015-02-18 19:59:58.846176300 -0500
@@ -218,9 +218,8 @@ with_threads=no
 with_gmalloc=no
 # i18n: essential for internationalized programs
 # syscalls: very useful for scripting
-# regexp: useful for scripting and text processing
 # readline: additional facilities for REPL, added later if appropriate
-BASE_MODULES="i18n syscalls regexp"
+BASE_MODULES="i18n syscalls"
 
 verbose=${CLISP_MAKEMAKE_VERBOSE:-false}
 # Handle --with-... arguments
@@ -2219,7 +2218,6 @@ if [ $HOS = unix ] ; then
 fi
 echol "# Add-on modules that are present in all linking sets (including base)"
 echol "# syscalls             export some POSIX (and other) system calls"
-echol "# regexp               POSIX regular expressions"
 echol "# i18n                 Lisp program internationalization"
 echol "# readline             extra fine REPL controls"
 echol "BASE_MODULES = ${BASE_MODULES}"
