diff -r a3b9fbbbd2dd src/pathname.d
--- a/src/pathname.d	Tue Mar 14 23:44:47 2017 -0400
+++ b/src/pathname.d	Wed Mar 22 09:06:45 2017 -0400
@@ -5383,7 +5383,7 @@
 	GC_SAFE_SYSTEM_CALL(result = readlink(namestring_asciz,linkbuf,linklen));
         if (result<0)
           OS_file_error(*(fs->fs_pathname));
-        if (result >= (int)linklen) { /* linkbuf too small - probably a link from /proc */
+        if (result > (int)linklen) { /* linkbuf too small - probably a link from /proc */
           FREE_DYNAMIC_ARRAY(linkbuf); linklen = 2*result; goto retry_readlink;
         }
         linklen = result;
